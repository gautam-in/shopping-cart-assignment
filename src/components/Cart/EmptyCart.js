import React from 'react';
import './style.css';



export default function EmptyCart(){
    return(
        <div>
        <div className="cart-no-item-title">
        <p>No items in your cart</p>
        <p className="cart-no-item-subtitle">Your favourite items are just a click away</p>
       
        </div>
        </div>

    )
}