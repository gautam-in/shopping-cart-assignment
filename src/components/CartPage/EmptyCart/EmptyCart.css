.empty-cart {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 68vh;
  background-color: #fff;
  flex: 1;
  position: relative;
}
.sub-container {
  display: flex;
  min-height: 40vh;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  flex: 1;
}
.emptyCart-title {
  font-size: 0.9rem;
}

.text {
  font-size: 0.8rem;
  padding: 1rem 0rem;
}

.footer {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 0 1rem;
  position: relative;
}
.startbutton {
  width: 100%;
}
.cartFooter {
  width: 92% !important;
}
