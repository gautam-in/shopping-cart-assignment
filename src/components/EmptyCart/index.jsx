import React from "react";
import { Button } from "../../global.styles";
import CartModalHeader from "../CartModalHeader";
import styled from "styled-components";

const EmptyCart = ({ setIsCartVisible }) => {
  return (
    <EmptyCartModal
      style={{
        display: "flex",
        flexDirection: "column",
        backgroundColor: "#fff",
      }}
    >
      <CartModalHeader quantity={0} setIsCartVisible={setIsCartVisible} />
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
          alignItems: "center",
          flex: 1,
        }}
      >
        <span style={{ fontSize: 20, fontWeight: 700 }}>
          No Items in your cart
        </span>
        <span>Your favourite items are just a click away</span>
      </div>
      <Button
        style={{
          display: "flex",
          width: "95%",
          alignSelf: "center",
          justifyContent: "center",
          marginBottom: 8,
        }}
        onClick={() => {
          setIsCartVisible((prev) => !prev);
        }}
      >
        Start Shopping
      </Button>
    </EmptyCartModal>
  );
};


const EmptyCartModal = styled.div`
  position: fixed;
  bottom: 0;
  z-index: 100;
  overflow: auto;
  display: flex;
  flex-direction: column;
  right: 10%;
  background-color: #fff;
  height: 85%;
  width: 35%;
`;

export default EmptyCart;
