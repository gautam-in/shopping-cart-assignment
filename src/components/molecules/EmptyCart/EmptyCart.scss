.empty-cart-body {
  background: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  h5 {
    font-size: 1.5rem;
  }
  p {
    padding-top: 1rem;
    font-weight: 600;
  }
}
.empty-cart-footer-wrap {
  padding: 2px 10px 10px 10px;
  background-color: white;

  p {
    font-size: 1.3rem;
    margin: 10px 0px;
    text-align: center;
    font-weight: 600;
  }
  .empty-cart-button {
    border-radius: 3px;
    height: 3.5rem;
  }
  .checkout-button {
    display: flex;
    justify-content: space-between;
    padding: 10px 10px;
    .arrow-checkout-icon {
      padding-left: 7px;
    }
  }
}

@media only screen and (max-width: 574px) {
  .empty-cart-body {
    height: calc(100% - 91px);
  }
}

@media only screen and (min-width: 576px) {
  .empty-cart-body {
    height: calc(100% - 99px);
  }
}

@media only screen and (min-width: 770px) {
  .empty-cart-body {
    height: calc(100% - 101px);
  }
}
