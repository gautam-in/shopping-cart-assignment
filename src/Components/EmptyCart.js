import React, {useState} from 'react';
import { Link } from 'react-router-dom';
import '../CSS/Cart.css'


const EmptyCart = ({onClick}) => {
    return (
        <div className='cartModal'>
            <header className='cartHeader'>
                <p>My Cart</p>
                <button onClick={onClick}>X</button>
            </header>
            <div className='emptyCart'>
                <span>No items in your cart</span>Your favourite items are just a
                click away
            </div>
            <div className='moveToShopping' onClick={onClick}>
                <Link to='/products'><button>Start Shopping</button></Link>
            </div>
        </div>
    )
}

export default EmptyCart;