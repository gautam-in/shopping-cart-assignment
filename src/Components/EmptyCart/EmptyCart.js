import Button from "../UI Components/Button/Button";
import "./EmptyCart.scss";

const EmptyCart = () => {
  return (
    <section className="empty-cart">
      <div className="empty-cart__container">
        <h2 className="empty-cart__container__title">No items in your cart</h2>
        <p className="empty-cart__container__text">
          Your favourite items are just a click away
        </p>
      </div>
      <footer className="empty-cart__footer">
        <Button className="empty-cart__footer__button">Start Shopping</Button>
      </footer>
    </section>
  );
};

export default EmptyCart;
