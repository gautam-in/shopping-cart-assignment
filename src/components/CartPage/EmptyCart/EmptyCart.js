import React, { useContext } from "react";
import "./EmptyCart.css";
import { useNavigate } from "react-router-dom";
import { BootstrapButton as Button } from "../../BootstrapButton";
import { GlobalContext } from "../../../context/GlobalContext";
import { Typography } from "@mui/material";

export default function EmptyCart() {
  //const browserWidth = window.matchMedia("(max-width: 700px)").matches;
  const navigate = useNavigate();
  const {
    cartItems: { cartOpen },
    dispatch,
  } = useContext(GlobalContext);

  const goToProducts = () => {
    dispatch({
      type: "HANDLE_CART",
      cartOpen: !cartOpen,
    });
    navigate("/products");
  };

  return (
    <section className="empty-cart">
      <div className="sub-container">
        <Typography variant="h6">No items in your cart</Typography>
        <Typography className="text">
          Your favourite items are just a click away
        </Typography>
      </div>
      <footer className="footer cartFooter">
        <Button className="startbutton" onClick={() => goToProducts()}>
          Start Shopping
        </Button>
      </footer>
    </section>
  );
}
