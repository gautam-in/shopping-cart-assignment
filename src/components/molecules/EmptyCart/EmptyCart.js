import React from 'react';
import './EmptyCart.scss';
import { useHistory } from 'react-router-dom';
import * as Constants from '../../../shared/constants';

const EmptyCart = ({ toggleCartModal }) => {
  const history = useHistory();
  return (
    <>
      <section className='empty-cart-body'>
        <h5 aria-label='No item in your cart'>{Constants.CartEmpty}</h5>
        <p aria-label='Your favourite items are just a click away'>{Constants.CartEmptyFavItems}</p>
      </section>
      <footer className='empty-cart-footer-wrap'>
        <button
          type='button'
          className='custom-button empty-cart-button'
          onClick={() => {
            toggleCartModal();
            history.push('/products');
          }}
        >
          {Constants.StartShopping}
        </button>
      </footer>
    </>
  );
};

export default EmptyCart;
