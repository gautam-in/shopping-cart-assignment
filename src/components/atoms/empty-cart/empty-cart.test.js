
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import EmptyCart from './index';

describe('App', () => {
    test('renders App component', () => {
        let component = render(<BrowserRouter><EmptyCart /></BrowserRouter>);
        expect(component).toMatchSnapshot();
        expect(screen.getByText('Start Shopping')).toBeInTheDocument();
        expect(screen.queryByText('Start Shopping22')).toBeNull();
        screen.getByRole('link');
    });
});