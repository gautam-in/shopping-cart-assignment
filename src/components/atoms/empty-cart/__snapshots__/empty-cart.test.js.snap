// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="empty-cart"
      >
        <div>
          No items in your cart
        </div>
        <div>
          Youy favourite items are just a click away
        </div>
        <div
          class="start-shop"
        >
          <a
            href="/products"
          >
            <button
              class=" button"
            >
              Start Shopping
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="empty-cart"
    >
      <div>
        No items in your cart
      </div>
      <div>
        Youy favourite items are just a click away
      </div>
      <div
        class="start-shop"
      >
        <a
          href="/products"
        >
          <button
            class=" button"
          >
            Start Shopping
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
